### PR TITLE
Add live

### PR DESCRIPTION
**Pull request by jamescyeh**

### CODE REVIEW

File: models/encoding.go [hunk 1]
```diff
@@ -309,3 +309,33 @@ type StatusResponse struct {
...
+	DashManifests []LiveDashManifest `json:"dashManifests,omitempty"`
+}
+
+type LiveStatusResult struct {
+	StreamKey *string `json:"streamKey,omitempty"`
+	EncoderIP *string `json:"encoderIp,omitempty"`
+}
+
+type LiveStatusData struct {
+	//Success fields
```
Comment (lukaskroepfl) [inline models/encoding.go]: Can't see comments in the other structs, we should probably remove this?

Comment (Tigraine) [inline models/encoding.go]: I'd leave these in - they don't hurt and in go comments are more commonplace :)

File: examples/live/create_simple_encoding_live_dash.go [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+			fmt.Println("---------------")
+			fmt.Println("Live Stream set up successfully:")
+			fmt.Printf("Encoding ID ... %v \n", *encodingResp.Data.Result.ID)
+			fmt.Printf("Encoder IP .... %v \n", *statusResp.Data.Result.EncoderIP)
+			fmt.Printf("Stream Key .... %v \n", *statusResp.Data.Result.StreamKey)
+			fmt.Printf("Stream URL: ... rtmp://%v/live \n", *statusResp.Data.Result.EncoderIP)
+			fmt.Println("---------------")
+			return
+		}
+		numRetries++
```
Comment (jamescyeh) [inline examples/live/create_simple_encoding_live_dash.go]: yeah make sense, I had a "while loop" before i should have chnaged this

File: examples/live/create_simple_encoding_live_dash.go [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+	}
+	fmt.Println("Maximum number of retries reached.")
+}
+
+func errorHandler(responseStatus bitmovintypes.ResponseStatus, err error) {
```
Comment (jamescyeh) [inline examples/live/create_simple_encoding_live_dash.go]: I will move this in a future small pr.

File: examples/live/create_simple_encoding_live_dash.go [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+		if err != nil {
+			if err.Error() != "ERROR 2023: Live encoding details not available!" {
+				fmt.Println("Error in starting live encoding")
+				fmt.Println(err)
+				return
+			}
+			fmt.Println("Encoding details not ready yet.")
+			numRetries++
+			continue
+		} else if statusResp != nil {
```
Comment (jamescyeh) [inline examples/live/create_simple_encoding_live_dash.go]: good catch!

**Comment by paulmacdougall** [comment]
Test

**Pull request merged**